src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -10,481 +10,42 @@
-                    ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.INVOICE_DETAIL_IDENTIFIER,
-                    totalCostInvoiceDetail.getInvoiceDetailIdentifier());
-            parameterMap.put(ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + KFSPropertyConstants.DOCUMENT_NUMBER,
-                    totalCostInvoiceDetail.getDocumentNumber());
-            parameterMap.put(ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.CATEGORIES,
-                    getConfigurationService()
-                            .getPropertyValueAsString(ArKeyConstants.CONTRACTS_GRANTS_INVOICE_DETAILS_TOTAL_LABEL));
-            parameterMap.put(ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.TOTAL_BUDGET,
-                    totalCostInvoiceDetail.getTotalBudget());
-            parameterMap.put(ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.INVOICE_AMOUNT,
-                    totalCostInvoiceDetail.getInvoiceAmount());
-            parameterMap.put(
-                    ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.CUMULATIVE_EXPENDITURES,
-                    totalCostInvoiceDetail.getCumulativeExpenditures());
-            parameterMap.put(ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.BUDGET_REMAINING,
-                    totalCostInvoiceDetail.getBudgetRemaining());
-            parameterMap.put(
-                    ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.TOTAL_PREVIOUSLY_BILLED,
-                    totalCostInvoiceDetail.getTotalPreviouslyBilled());
-            parameterMap.put(ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.ESTIMATED_COST,
-                    totalCostInvoiceDetail.getTotalPreviouslyBilled().add(totalCostInvoiceDetail.getInvoiceAmount()));
-            parameterMap.put(
-                    ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.TOTAL_AMOUNT_BILLED_TO_DATE,
-                    totalCostInvoiceDetail.getTotalAmountBilledToDate());
-            parameterMap.put(
-                    ArPropertyConstants.TOTAL_INVOICE_DETAIL + "." + ArPropertyConstants.AMOUNT_REMAINING_TO_BILL,
-                    totalCostInvoiceDetail.getAmountRemainingToBill());
-            parameterMap.put(
-                    CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.TOTAL_PROGRAM_OUTLAYS_TO_DATE,
+            Map<String, Object> parameterMap =  new HashMap<String, Object>();
+            
+            parameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.TOTAL_PROGRAM_OUTLAYS_TO_DATE,
                     totalCostInvoiceDetail.getTotalAmountBilledToDate().add(document.getInvoiceGeneralDetail().getCostShareAmount()));
+            
+            Map<String, String> localParameterMap = new PdfFormattingMap(parameterMap);
+            templateParameters.putAll(new PdfFormattingMap(localParameterMap));
```

Comment:
The sub-map is getting wrapped inside PdfFormattingMap twice by the code above. I would suggest wrapping it inside only one PdfFormattingMap before passing it to the main map.

Alternatively, you could convert the total-program-outlays-to-date value to a String directly and then pass it to the main map. This service should already have a variable for the ContractsGrantsBillingUtilityService, which is needed for performing the proper KualiDecimal-to-String conversion that PdfFormattingMap does.

Reply:
ah, I will fix.  miss step from refactoring.  I'd like to continue to use PdfFormattingMap so the string conversions are consistent.

Reply:
Sounds good. Thanks!